test/jdk/java/awt/Frame/ShapeNotSetSometimes/ShapeNotSetSometimes.java

```diff
@@ -124,48 +125,49 @@ public static void main(String[] args) throws Exception {
     }
 
     private void doTest() throws Exception {
-        Point wls = backgroundFrame.getLocationOnScreen();
+        EventQueue.invokeAndWait(backgroundFrame::toFront);
+        robot.waitForIdle();
 
-        robot.mouseMove(wls.x + 5, wls.y + 5);
-        robot.mousePress(InputEvent.BUTTON1_DOWN_MASK);
-        robot.delay(10);
-        robot.mouseRelease(InputEvent.BUTTON1_DOWN_MASK);
-        robot.delay(500);
+        EventQueue.invokeAndWait(window::toFront);
+        robot.waitForIdle();
 
-        EventQueue.invokeAndWait(window::requestFocus);
+        robot.delay(500);
 
-        robot.waitForIdle();
         try {
-            Thread.sleep(300);
-        } catch (InterruptedException e) {
-            // ignore this one
-        }
-
-        // check transparency
-        final int COUNT_TARGET = 10;
-
-        // checking outside points only
-        for(int i = COUNT_TARGET; i < COUNT_TARGET * 2; i++) {
-            int x = pointsToCheck[i][0];
-            int y = pointsToCheck[i][1];
-            boolean inside = i < COUNT_TARGET;
-            Color c = robot.getPixelColor(window.getX() + x, window.getY() + y);
-            System.out.println("checking " + x + ", " + y + ", color = " + c);
-            if (inside && BACKGROUND_COLOR.equals(c) || !inside && !BACKGROUND_COLOR.equals(c)) {
-                System.out.println("window.getX() = " + window.getX() + ", window.getY() = " + window.getY());
-                System.err.println("Checking for transparency failed: point: " +
-                        (window.getX() + x) + ", " + (window.getY() + y) +
-                        ", color = " + c + (inside ? " is of un" : " is not of ") +
-                        "expected background color " + BACKGROUND_COLOR);
-                throw new RuntimeException("Test failed. The shape has not been applied.");
+            colorCheck(innerPoint.x, innerPoint.y, SHAPE_COLOR);
+
+            for (Point point : pointsOutsideToCheck) {
+                colorCheck(
+                        point.x,
+                        point.y,
+                        BACKGROUND_COLOR
+                );
```

Comment:
Dunno why the old version was not checking inner points, but I've added one just to check if the shaped frame is above background one.

Reply:
Don't we need to check the places where the shadow is rendered that it is not the shape color(to prove that the shape is not simple rectangular)?

Reply:
Sure, extra check added.